Cubical/Structures/QuotientRing.agda

```diff
@@ -0,0 +1,248 @@
+{-# OPTIONS --cubical --safe #-}
+
+module Cubical.Structures.QuotientRing where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Equiv
+open import Cubical.Foundations.HLevels
+open import Cubical.HITs.SetQuotients.Base
+open import Cubical.HITs.SetQuotients.Properties
+open import Cubical.Structures.Ring
+open import Cubical.Structures.Ideal
+
+
+private
+  variable
+    ℓ : Level
+
+module _ (R′ : Ring {ℓ}) (I : ⟨ R′ ⟩  → hProp ℓ) (I-isIdeal : isIdeal R′ I) where
+  private
+    open ring-syntax R′
+    open isIdeal I-isIdeal
+    open ring-axioms R′
+    open calculations R′
+    R = ⟨ R′ ⟩
+
+  R/I : Type ℓ
+  R/I = R / (λ x y → x - y ∈ I)
+
+  private
+    -isDistributive : (x y : R) →  (- x) + (- y) ≡ - (x + y)
+    -isDistributive x y =
+      implicitInverse _ _
+           ((x + y) + ((- x) + (- y)) ≡⟨ sym (ring+-assoc _ _ _) ⟩
+            x + (y + ((- x) + (- y))) ≡⟨ cong
+                                           (λ u → x + (y + u))
+                                           (ring+-comm _ _) ⟩
+            x + (y + ((- y) + (- x))) ≡⟨ cong (λ u → x + u) (ring+-assoc _ _ _) ⟩
+            x + ((y + (- y)) + (- x)) ≡⟨ cong (λ u → x + (u + (- x)))
+                                              (ring+-rinv _) ⟩
+            x + (₀ + (- x))           ≡⟨ cong (λ u → x + u) (ring+-lid _) ⟩
+            x + (- x)                 ≡⟨ ring+-rinv _ ⟩
+            ₀ ∎)
+
+    translatedDifference :
```

Comment:
I guess this should also be a general lemma? Maybe all of these can go into a module "theory" in the Ring file?

Reply:
There is already a module 'calculations' in the ring file, so I renamed it to 'theory'. And I moved the lemma to that module.